docs/proposal-multi-user-approach.md

```diff
@@ -36,156 +36,174 @@ UI Server back-end code should prevent users manipulating client-side code to by
+    "<user1>": ["READ", "pause", "trigger", "message"], # Specified interactions allowed
+    "group:<group1>": ["ADMIN"],                # All access to workflows granted
+    "<user2>": ["ADMIN", "!trigger", "!edit"],  # All Admin, not trigger, or edit
+    "<user3>": ["!ADMIN"]                       # No privileges (incl READ)
 }
 ```
 
 ### Example Site Configuration
 
 ```python
 c.UIServer.authorisation = {
-    {
-        "<*>": {                    # For all ui-server owners,
-            "<*>": {                # Any authenticated user
-                "default": "read",  # Will have default read-only access
-                "limit": "execute"  # All server owners are allowed to raise access up to
-                                    # a maximum of execute.
-            }
-        },
-        "<server_owner1>": {        # Specific UI Owner
-            "<user1>": {            # Specific user1
-                "limit": "write"    # Can only be granted a maximum of write by
-                                    # server_owner1
-            },
-            "group:<groupA>": {"default": "write"},  # Denote a group with `group:`
+    "<*>": {                    # For all ui-server owners,
+        "<*>": {                # Any authenticated user
+            "default": "READ",  # Will have default read-only access
+            "limit": "READ"     # All server owners default to being able to give away
+        },                      # a maximum of read privileges.
+        "<user1>": {            # user1
+            "default": "!ADMIN" # No privileges for all ui-server owners
+        },                      # No limit set, default level used as limit i.e. all
+    },                          # ui server owners default to be able to give away a
+                                # maximum of no privileges to user1
+
+    "<server_owner_1>": {       # For specific UI Server owner,
+        "<*>": {                # Any authenticated user
+            "default": "READ",  # Will have default read-only access
+            "limit": "ADMIN"    # server_owner_1 is able to give away
+        },                      # a maximum of admin privileges.
+    },
+    "<server_owner_2>": {       # Specific UI Owner
+        "<user2>": {            # Specific user2
+           "limit": "ADMIN"     # Can only be granted a maximum of Admin by
+        },                      # server_owner1, default access for user2 falls back to
+                                # standard READ only (if server_owner_2/user2 are
+                                # included in other auth config e.g. the top example),
+                                # or none if not in any other auth config sections.
+
+        "group:<groupA>": {     # group denoted with a `group:`
+            "default": "ADMIN"  # groupA has default ADMIN access to server_owner_2's
+        },                      # workflows
+    },
+    "group:<grp_of_svr_owners>":{         # Group of users who own UI Servers
```

Comment:
I am not sure of the business need for this, although included it for discussion. Do we want to support making authorisation by a group of UI Server owners possible? Perhaps this is unnecessary complexity?

Reply:
Struggling to think of the case here too. Is it something like this?: "oper" users can authorize "test" users to access any oper server?

Reply:
Yes, that is what I had envisioned, although unsure if that would be a useful feature. I guess it depends if that kind of bulk permission for authorisation/default settings has a use case.

Reply:
I think it's a reasonable feature to have and hopefully not too difficult since we need to support groups for specifying the users. One use case: all members of the group "my-team" grant READ access to other members of "my-team".